src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -78,27 +82,42 @@ public static void apartmentToJson(Apartment a, Path jsonPath) throws IOExceptio
 
 	/**
 	 * Gets the address field from an Address JSON.
+	 * 
+	 * We have to make sure that we have everything at every step <br>
+	 * It is for this reason that the code is filled with check
+	 * <p>
+	 * The contract for this API is returned either <code>"features": []<code> if
+	 * the address cannot be found or in shape <code>{"features": ["properties":
+	 * {"label": "AdressHere"}]}<code>
+	 * </p>
 	 *
-	 * @param jsonString <i>String</i> the Address into JSON format
-	 * @return <i>String</i> the address field
+	 * @param jsonString {@link String} the Address into JSON format
+	 * @return {@link String} the address field
+	 * 
+	 * @throws IllegalArgumentException in the case where the deserialization of the
+	 *                                  JSON encounters a problem
 	 */
-	public static String getAddressFromJson(String jsonString) {
-		Jsonb jsonb = JsonbBuilder.create();
-
-		final LinkedHashMap<?, ?> result = jsonb.fromJson(jsonString, LinkedHashMap.class);
-		LOGGER.info("Get address");
-
-		checkArgument(!result.get("address").toString().isEmpty() && result.containsKey("address"),
-				"There is no field adress in the JSON file.");
-		return result.get("address").toString();
+	public static String getAddressFromJson(String jsonString) throws IllegalArgumentException {
+		checkArgument(jsonString != null, "jsonString cannot be empty");
```

Comment:
cf guava : checkNotNull

Reply:
J'avais mis une checkNotNull mais le problème c'est que ça ne lèvera pas une  `IllegalArgumentException`. Or, `ApartmentFactory` récupère uniquement les `IllegalArgumentException`

Reply:
Si la chaine récupéré est null on est sur un problème différents que celui de la chaine n'ayant pas d'adresse. On arrive sur le fait que l'API ne retourne pas de données et le système de retry a déjà été refusé.

Reply:
Après avoir testé sur + de 500 adresses, je confirme que l'API ne renvoi, dans son cas nominal, de chaine vide

Reply:
Fix in f928f39f367274a146722e569fb63e3c5a84b617
Finalement `getAddressFromJson()` lève une exception `InvalidObjectException` uniquement si nous devons réitérer. Dans le cas ou une `InvalidObjectException` est levé cela veut dire que nous avons :
```json
{"type": "FeatureCollection", "version": "draft", "features": [], "attribution": "BAN", "licence": "ETALAB-2.0", "limit": 1}
```